app/views/courses/TeacherCoursesView.coffee

```diff
@@ -66,4 +68,10 @@ module.exports = class TeacherCoursesView extends RootView
     language = form.find('.language-select').val()
     window.tracker?.trackEvent 'Classes Guides Play Level', category: 'Teachers', courseID: courseID, language: language, levelSlug: levelSlug, ['Mixpanel']
     url = "/play/level/#{levelSlug}?course=#{courseID}&codeLanguage=#{language}"
-    application.router.navigate(url, { trigger: true })
+    firstLevelSlug = @campaigns.get(@courses.at(0).get('campaignID')).getLevels().at(0).get('slug')
+    if levelSlug is firstLevelSlug
+      @listenTo @openModalView(@heroSelectModal),
+        'hide': ->
+          application.router.navigate(url, { trigger: true })
+    else
+      application.router.navigate(url, { trigger: true })
```

Comment:
This looks Promise-worthy to me. What do you think of this? Throw this on `CocoView`:

```    
  wait: (event) ->
    return new Promise((resolve) => @once(event, resolve))
```

Then here:

```
if levelSlug is firstLevelSlug
  promise = @openModalView(@heroSelectModal).wait('hide')
else
  promise = Promise.resolve()
promise.then -> application.router.navigate(url, { trigger: true })
```

Or something along those lines.